include/mpm_explicit.tcc

```diff
@@ -454,6 +462,37 @@ bool mpm::MPMExplicit<Tdim>::apply_nodal_tractions() {
   return status;
 }
 
+//! Assign materials to particle sets
+template <unsigned Tdim>
+bool mpm::MPMExplicit<Tdim>::apply_entity_sets_properties() {
+  bool status = false;
+  // Assign material to particle sets
+  try {
+    // Get particle properties
+    auto particle_props = io_->json_object("particle");
+    // Get particle sets properties
+    auto particle_sets = particle_props["particle_sets"];
+    // Assign material to each particle sets
+    for (const auto& psets : particle_sets) {
+      // Get set material from list of materials
+      auto set_material = materials_.at(psets["material_id"]);
+      // Get sets ids
+      std::vector<mpm::Index> sids = psets["set_id"];
```

Comment:
I'm confused as to what's happening here. Could you show me the JSON object for this?